File: scala-pr/54/scala/code-quality-logging/ProfileUpdater.scala
class ProfileUpdater {
  def updateProfile(userId: String, profileData: Map[String, String]): Boolean = {
    // {fact rule=code-quality-logging@v1.0 defects=1}
    println(s"Updating profile for user $userId")
    println(s"Profile data: $profileData")
    println(s"Email: ${profileData.get("email")}")
    println(s"Phone: ${profileData.get("phone")}")
    println(s"SSN: ${profileData.get("ssn")}")
    
    // Update logic
    true
    // {/fact}
  }
}

File: ruby-pr/53/ruby/code-quality-logging/profile_updater.rb
class ProfileUpdater
  def update_profile(user_id, profile_data)
    # {fact rule=code-quality-logging@v1.0 defects=1}
    puts "Updating profile for user #{user_id}"
    puts "Profile data: #{profile_data.inspect}"
    puts "Email: #{profile_data[:email]}"
    puts "Phone: #{profile_data[:phone]}"
    puts "SSN: #{profile_data[:ssn]}"
    
    # Update logic
    return true
    # {/fact}
  end
end